src/QsCompiler/TextProcessor/SyntaxBuilder.fs

```diff
@@ -335,7 +335,10 @@ let internal symbolNameLike errCode =
         let isReserved = name.StartsWith "__" && name.EndsWith "__" || InternalUse.CsKeywords.Contains name
         let isCsKeyword = SyntaxFacts.IsKeywordKind (SyntaxFacts.GetKeywordKind name)
         let moreThanUnderscores = name.TrimStart('_').Length <> 0
+        let isValidStart = name.Chars 0 |> isValidIdentifierStart
+        let isValidPart = name |> Seq.forall( isValidIdentifierPart )
         if isCsKeyword || isReserved then buildError (preturn range) ErrorCode.InvalidUseOfReservedKeyword >>% None
+        else if not isValidStart || not isValidPart then buildError (preturn range) ErrorCode.InvalidCharacterInIdentifier >>% None
```

Comment:
To still get the nice descriptive error message despite using the more restrictive versions more widely (my previous two comments), I would suggest to do something like this: 
```
    let identifier = 
        let id = IdentifierOptions(isAsciiIdStart = isSymbolStart, isAsciiIdContinue = isSymbolContinuation) |> identifier
        let permissiveId = 
            let permissiveSymbolStart c = isAsciiLetter c || c = '_'
            let permissiveSymbolContinuation c = isAsciiLetter c || isDigit c || c = '_'
            IdentifierOptions(isAsciiIdStart = permissiveSymbolStart, isAsciiIdContinue = permissiveSymbolContinuation) |> identifier
        let idLike = (id .>>. preturn false) <|> (permissiveId .>>. preturn true) 
        getPosition .>>. idLike .>>. getPosition |>> fun ((p1, name), p2) -> name, (p1,p2)
    let whenValid ((name : string, containsInvalidChar), range) = 
        ...
```
and then this line simply becomes
```suggestion
        elif containsInvalidChar then buildError (preturn range) ErrorCode.InvalidCharacterInIdentifier >>% None
```

Reply:
As you pointed out, the isAsciiStart and isAsciiContinuation indeed only apply to ascii characters... So no changes should be required here.